lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1030,6 +1063,37 @@ SwiftLanguageRuntimeImpl::GetNumChildren(CompilerType type,
   if (auto *eti = llvm::dyn_cast_or_null<swift::reflection::EnumTypeInfo>(ti)) {
     return eti->getNumPayloadCases();
   }
+  // Objects.
+  if (auto *rti =
+      llvm::dyn_cast_or_null<swift::reflection::ReferenceTypeInfo>(ti)) {
+
+    switch (rti->getReferenceKind()) {
+    case swift::reflection::ReferenceKind::Weak:
+    case swift::reflection::ReferenceKind::Unowned:
+    case swift::reflection::ReferenceKind::Unmanaged:
+      // Weak references are implicitly Optionals, so report the one
+      // child of Optional here.
+      if (GetWeakReferent(*ts, type))
+        return 1;
+      break;
+    default:
+      break;
+    }
+
+    if (!tr)
+      return {};
+
+    auto *reflection_ctx = GetReflectionContext();
```

Comment:
Would this fit nicely into one case in the switch above, or are there many different kinds for which this path makes sense?

Reply:
good question, I'm not sure yet. Once this PR is working, I'll compare the paths then.

Reply:
What do you think? There are a couple distinct conditions in each case that don't make sense to the other case, and so combing the two doesn't feel like a no-brainer to me.

Reply:
I was just thinking of writing this as 

```
    switch (rti->getReferenceKind()) {
     case swift::reflection::ReferenceKind::Weak:
     case swift::reflection::ReferenceKind::Unowned:
     case swift::reflection::ReferenceKind::Unmanaged:
       // Weak references are implicitly Optionals, so report the one
       // child of Optional here.
       if (GetWeakReferent(*ts, type))
         return 1;
       break;
     default: {
       if (!tr)
         return {};
       auto tc = swift::reflection::TypeConverter(reflection_ctx->getBuilder());
....
```
but that's purely cosmetic and not important at all...

Reply:
thanks, I had misunderstood. I will make that change in a follow up.